Protocol.md

```diff
@@ -66,6 +66,7 @@ the funding transaction and CETs.
 
 1. type: 42778 (`offer_dlc_v0`)
 2. data:
+   * [`protocol_version`: `bigsize`]
    * [`byte`:`contract_flags`]
```

Comment:
seems a little redundant to have both of these

Reply:
I personally think the offer should signal the features it supports and requires which are two separate things, and then have a field in the accept with the feature set to be used, so for example you could have an offer that had a new feature enabled but not required where the accepter can choose whether or not to use it.

Reply:
I personally see these two fields as providing different information.
`protocol_version` indicates which language I speak, and enables the receiver to determine whether they will be able to understand the content of the messages that are exchanged. Ideally in the future nodes should advertise which version of the protocol they support so that other peers will know how to communicate with them.
`contract_flags` are used to signal features as you mention.
Does that make sense?

Reply:
In a Lightning/DLC world, I'm not sure if the `protocol_version` is making sense as new protocol subset can be introduced as new messages types and the question of whom to talk to is already solved with the feature bits space ?
I agree with the necessity of a `contract_flags`, see my other point mentioning the LN discussion on `channel_type`.

Reply:
For me `contract_flags` ad `protocol_version` are two separate things. The protocol version is for breaking changes in the protocol specifications and enabling nodes to see whether they are compatible with a version or not. Contract flags is for a given protocol version whether a node supports a feature or not.
I guess we could use contract version for both but it doesn't feel optimal to me.
Happy to discuss though, I don't have that strong of an opinion about how things should be, just that it's practical to have some sort of upgradeability mechanism.